### DISCUSSION TITLE
android 下如何实现PAG动效的透明化

### DISCUSSION THREAD

**Discussion by chesterwan1** [Tencent/libpag#1886] [Q&A] [answered]
设计侧设计并给出了一个透明的PAG资源，android侧按照下面的方式进行加载：
       mGlSurfaceView = (GLSurfaceView) rootView.findViewById(R.id.wiveView1);
        mGlRender = new GLRender(getContext());
        mGlSurfaceView.setEGLContextClientVersion(2);
        mGlSurfaceView.setEGLConfigChooser(8, 8, 8,
                8, 16, 0); // 设置EGLConfig以支持透明度
        mGlSurfaceView.getHolder().setFormat(PixelFormat.TRANSLUCENT); // 设置为半透明
        mGlSurfaceView.setRenderer(mGlRender);
        mGlSurfaceView.setRenderMode(GLSurfaceView.RENDERMODE_CONTINUOUSLY);
GLRender 中设置并加载PAG文件：
GLES20.glClearColor(0.5f, 0.5f, 0.5f, 0f);
 PAGFile pagFile = PAGFile.Load(context.getAssets(), "flow_sound_effect_surround_up_left.pag");
但是实际效果并不是透明的，问题出在哪儿呢，附件是对应的pag文件
实际效果是：
![image](https://github.com/Tencent/libpag/assets/151141280/f7953ea9-6c93-4989-935b-95517e6dee3b)

[file.zip](https://github.com/Tencent/libpag/files/13390121/file.zip)

**Comment by kevingpqi123** [comment, marked as answer]
你可以先搜索下 GLSurfaceView 如何显示透明的 WebP 图片，这个地方实现了，pag 的透明显示基本上也就实现了
